### PR TITLE
Implementation of Boundary Conditions

### PR DESCRIPTION
**Pull request by twicki**
## Technical Description

These are the (minor) changes in functionality to implement boundary conditions into gtclang. The bulk of the boundary-conditons are in  https://github.com/MeteoSwiss-APN/dawn/pull/49

- Parsing of Boundary Conditions by the stencil-parser
- Implementation of the Preprocessors rewriting
- Two simple tests

### CODE REVIEW

File: src/gridtools/clang/storage_runtime.hpp [hunk 1]
```diff
@@ -94,5 +94,14 @@ namespace gridtools {
         /** @} */
 
         /** @} */
+
```
Comment (twicki) [inline src/gridtools/clang/storage_runtime.hpp]: This feels like a bad way of doing it. Do you have a better idea?

Comment (twicki) [reply src/gridtools/clang/storage_runtime.hpp]: We choose to leave it as is since once the newest gridtools boudary-condition syntax rolls out we need to rework this anyways (and this will presumably be soon)
